### PR TITLE
remove void* casts

### PR DESCRIPTION
**Pull request by DaveBerkeley**

### CODE REVIEW

**Comment by DaveBerkeley** [comment]
Sorry, we are both making edits here!

**Comment by DaveBerkeley** [comment]
now has a double free. I'll fix it.

File: example/tb.cpp [hunk 1]
```diff
@@ -56,13 +56,13 @@ int main(int argc, char **argv, char **env) {
     "gpio.SW14",
     "gpio.SW15",
   };
-  int num_inputs = 16;
+  int num_inputs = sizeof(inputs) / sizeof(inputs[0]);
```
Comment (olofk) [inline example/tb.cpp:59]: Does this require all strings to be of equal length?

Comment (olofk) [reply example/tb.cpp:59]: Ohh.... using sizeof on an array gets the size of the index pointers. Is that so?

Comment (DaveBerkeley) [reply example/tb.cpp:59]: total size / sizeof of one element gives number in a array. Assuming no padding, but were are talking pointers here

**Comment by olofk** [comment]
Many thanks! Looks all good. Picked and pushed

**Pull request merged**